### PR TITLE
Blob meat, blob food, blob drinks, blob reagents

### PR DESCRIPTION
**Pull request by Securitymain**
[WIP][needspritework][gameplay]
## What this does

Makes blob nodes drop blob meat when hit with a sharp weapon.
A weapon with force 8 and sharpness 0.5 has a drop chance of 4%. This is rolled every time the blob node is hit.
This chance applies to shovels, a weapon like the butcher's knife will have a chance of 18%.
Blob cores drop a special type of meat, currently only used in one recipe.

Adds the blobanine reagent, gained from blob meat. Used in a number of recipes. When ingested, colors eyes green.

Blob meat is infected with the blob spore disease, transmitted by eating the meat.

Adds a line of blob food:
grafik

Adds a line of blob drinks:
grafik

Special "food":
Blob volcano/bleb. Gained by using blob core meat in a recipe. It's a structure like the poutine ocean.
- Spawns containing 30 generic blob meat pieces. Explodes the meat all over shortly after spawning
- Can be fed reagents on harm intent (food is also reagent_containers, so this is required...)
- Turns 10 nutriment into 1 blob meat
- Turns 5 mutagen into 1 friendly blob spore
- Can be fed food items
- Can be kicked to force an eruption, tosses whatever food items it contains in a 3x3 range randomly
- I don't really like this thing and the sprite sucks. Feedback please
- Earlier version of it, sprite changed since then: https://streamable.com/uwei92

Special drink:
Spore, requires karmotrine and blobanine
- Drinking this grants you the power to summon 2 weaker spores until it either fully metabolizes or you used up all of it.
- Costs 5 "spore" on use. Metabolizes slowly.
- Spores have 1/2 the health and size of regular ones. Pack the same punch. They can fly over tables and mobs.
- 30 second cooldown

Showing friendly spores spawned by the volcano and how the spore power works: https://streamable.com/3j0nmu

The food & drinks turned out good but I'm not so sure about thhe blob volcano or the spore drink

----------
Once this is done, will work on extending science biology 6+ with blob/bio themed items
Also another secret idea pertaining to ingesting core essences

Special thanks for spritework goes to
@falcon2346 
@Blithering 
sage of fish
Kurfursten

## Why it's good

Adds more crew interactivity with blobs and a reason to stay during a blob round. Incentivizes getting up close and personal with the blob instead of just shooting it with liberators before calling it. More in a later PR, probably.

## Changelog

:cl:
 * rscadd: Blob nodes drop meat, blob food, drinks & reagents
 * imageadd: Blob food & drinks

### CODE REVIEW

**Comment by Securitymain** [comment]
Remind myself to change the meat drop factor to 1 when it's done

**Comment by angryturnip** [comment]
Please make sure a blob kebab is added. It's a good item just for the name

**Comment by Securitymain** [comment]
Post more food & drink ideas

**Comment by Hinaichigo** [comment]
blob butter as a butter substitute

**Comment by Eneocho** [comment]
Blob ranching ISREAL??? WITH OXYGEN BLOB DOES EVOLUATE????
You could add blob meat to the exotic meats pool for the trophy meats crate.

**Comment by ghost** [comment]
Mountain Dew Blobja Blast

**Comment by wotisjanitor** [comment]
blob loaf

**Comment by Eneocho** [comment]
Requesting an admin only "burger that makes you into a blob" that makes you into a blob upon biting it.

**Comment by DeityLink** [comment]
>none of the foodstuff contains any blob spores

come the fuck ON!

**Comment by Eneocho** [comment]
You say that but if the blobfood had had blob spores in the first place the metaclub'd be here screaming bloody self-antagging.

**Comment by Securitymain** [comment]
Still havnt looked at blob spore stuff, getting the easy boilerplate recipes in first

**Comment by DeityLink** [comment]
> You say that but if the blobfood had had blob spores in the first place the metaclub'd be here screaming bloody self-antagging.

I'm not talking about stuff that turns you into an overmind, but about the pathogen that causes you to spread some blob when you die (but doesn't kill you on its own).

> Still havnt looked at blob spore stuff, getting the easy boilerplate recipes in first

Alright all you need to do when spawning the meat is:

```
if (!(source.looks in blob_diseases))
	CreateBlobDisease(source.looks)
var/datum/disease2/disease/D = blob_diseases[source.looks]
```
(`source` being the node tile, so you gotta call that BEFORE deleting it) to spawn the disease, then call `infect_disease2(D)` on your meat.

Cooking items in microwaves removes pathogen I believe so that leaves only the raw blob meat as a potential health hazard, which would also be a boon to the Virologist as a more straightforward way to acquire the pathogen.

**Comment by Securitymain** [comment]
nice thanks

**Comment by Kurfursten** [comment]
lib

We're returning fire!

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -1569,6 +1569,73 @@
 	reagents.add_reagent(NUTRIMENT, 8)
 	bitesize = 2
 
+/obj/item/weapon/reagent_containers/food/snacks/blobburger
+	name = "bloburger"
+	desc = "Careful, has a tendency to spill sauce in every direction when squeezed too hard."
```
Comment (Eneocho) [inline code/modules/reagents/reagent_containers/food/snacks.dm:1574]: Could use a chance for the crumbs to be dribbles instead, with that description.

**Comment by Eneocho** [comment]
Requesting a small chance for the evoluator drink to make you say "It really did evoluate!" or similar.

**Comment by Securitymain** [comment]
I'll put the dribbles in, for the say thing it's a maybe

**Comment by Securitymain** [comment]
No idea why the icondiffbot says I modified some of the right-hand in-hand sprites, I went as far as downloading the current file off this repo and re-adding the new drinks by hand and it still did it. Repo is up to date too... weird.

It's ready for review @Kurfursten

**Comment by pedr69** [comment]
the microwave kills the blob spores so the cooked meals aren't infected, right?

**Comment by Securitymain** [comment]
yes

**Pull request merged**